sdk/servicebus/service-bus/src/track2/serviceBusReceiverClient.ts

```diff
@@ -0,0 +1,568 @@
+      session = sessionOrOptions3;
+      options = options4 || {};
+    } else {
+      options = sessionOrOptions3 || {};
+    }
+
+    const { context, entityPath } = createConnectionContext(auth1, options);
+    this._context = context;
+    this._receiveMode = convertToInternalReceiveMode(receiveMode2);
+
+    const clientEntityContext = ClientEntityContext.create(
+      entityPath,
+      ClientType.ServiceBusReceiverClient,
+      context,
+      `${entityPath}/${generate_uuid()}`
+    );
+
+    // TODO: use the session connections object to "cache" the client entity context
+    if (session != null) {
+      const receiver = new InternalSessionReceiver(clientEntityContext, this._receiveMode, {
+        sessionId: session.id
+      });
+      this._sessionEnabled = true;
+      this._receiver = receiver;
+
+      this.diagnostics = {
+        async peek(maxMessageCount?: number): Promise<ReceivedMessage[]> {
+          return (await receiver.peek(maxMessageCount)).map((m) => m as ReceivedMessage);
+        },
+        async peekBySequenceNumber(
+          fromSequenceNumber: Long,
+          maxMessageCount?: number
+        ): Promise<ReceivedMessage[]> {
+          return (await receiver.peekBySequenceNumber(fromSequenceNumber, maxMessageCount)).map(
+            (m) => m as ReceivedMessage
+          );
+        }
+      };
+    } else {
+      const receiver = new InternalReceiver(clientEntityContext, this._receiveMode);
+      this._sessionEnabled = false;
+      this._receiver = receiver;
+
+      this.diagnostics = {
+        async peek(maxMessageCount?: number): Promise<ReceivedMessage[]> {
+          return (await receiver.peek(entityPath, maxMessageCount)).map(
+            (m) => m as ReceivedMessage
+          );
+        },
+        async peekBySequenceNumber(
+          fromSequenceNumber: Long,
+          maxMessageCount?: number
+        ): Promise<ReceivedMessage[]> {
+          return (
+            await receiver.peekBySequenceNumber(entityPath, fromSequenceNumber, maxMessageCount)
+          ).map((m) => m as ReceivedMessage);
+        }
+      };
+    }
+  }
+
+  /**
+   * Streams messages to the passed in handlers.
+   * @param handlers message handlers that receive events as well as errors.
+   */
+  subscribe(handlers: MessageHandlers<ContextWithSettlement>, options?: SubscribeOptions): void;
+  /**
+   * Streams messages to the passed in handlers.
+   * @param handlers message handlers that receive events as well as errors.
+   */
+  subscribe(handlers: MessageHandlers<{}>, options?: SubscribeOptions): void;
+  subscribe(
+    handlers: MessageHandlers<{}> | MessageHandlers<ContextWithSettlement>,
+    options?: SubscribeOptions
+  ): void {
+    // TODO: use options
+    if (this._receiveMode === ReceiveMode.peekLock) {
+      const onMessage = async (sbMessage: ServiceBusMessage) => {
+        await handlers.processMessage(sbMessage, settlementContext);
+      };
```

Comment:
Nit: You don't have to use async/await here:
```suggestion
      const onMessage = (sbMessage: ServiceBusMessage) => {
        return handlers.processMessage(sbMessage, settlementContext);
      };
```

Reply:
You can just paraphrase as UUOA (like UUOC)